### ISSUE TITLE
Speed drastically slows down after the first epoch

### ISSUE THREAD

**Issue by richarddwang** [PyTorchLightning/pytorch-lightning#12924]
## 🐛 Bug

Training speed (it/s) get 2x slower after the first epoch, 
and it seems to get slower after every epoch.

This observation appears when upgrading from 1.5.10 to 1.6.0
( i.e. 1.5.10 works well and is ok )

### To Reproduce

I can confirm it with both my code and this colab notebook
https://colab.research.google.com/drive/18NgixZRZR5i6L6xDU6P1dPMBoizq1A7i?usp=sharing (edit: sharing setting is set)

cc @borda @akihironitta

**Comment by DonkeyShot21** [comment]
Having the same issue, it slows down a lot after the first epoch. Versions before 1.6.0 seem to work fine.

**Comment by xieyin666** [comment]
Hi, my model takes up a lot of video memory and needs multi-gpu training. I set parameters according to the official example, but kept getting errors. Do you have any good tips to share with me? Thank you very much! My code setup is as follows:
![image](https://user-images.githubusercontent.com/83076581/166092233-2d15556e-b7d1-46cc-814a-988df4a804bb.png)
![image](https://user-images.githubusercontent.com/83076581/166092265-8830dfca-d71c-40e3-9a4f-aec902014445.png)

**Comment by dududukekeke** [comment]
Hi, my model takes up a lot of video memory and needs multi-gpu training. I set parameters according to the official example, but kept getting errors. Do you have any good tips to share with me? Thank you very much! My code setup is as follows:

![image](https://user-images.githubusercontent.com/83076581/166092448-129905c7-6f0a-4322-828c-f6af380729be.png)
![image](https://user-images.githubusercontent.com/83076581/166092458-70464a71-ecab-4d02-8bb8-77d6f9c6b6f7.png)

**Comment by DonkeyShot21** [comment]
I was using multiple GPUs with DDP.

**Comment by carmocca** [comment]
Using the code above but setting

```python

        accelerator="gpu",
        devices=4,
        strategy="ddp",
```

gives me


| version | Epoch 0 | Epoch 1 | Total   |
|---------|---------|---------|---------|
| 1.5.10  | 4.039  | 3.174  | 10.206 |
| 1.6.3   | 3.962  | 3.112  | 10.051 |
| master  | 3.834  | 3.046  | 9.986 |

**Comment by richarddwang** [comment]
Hi all,

Confirmed by my own code and the notebook above,
I found that the issue has been resolved in 1.6.3.

For observing the issue, you can install 1.6.0 <= version <= 1.6.2 instead.

Anyway, since it works now.
I am going to close the issue.
